### PR TITLE
Offline kinesis with retry on connect failure

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Introduce mecanism to retry kinesis readable creation after some delay.
Currently if missing program stays in zombie state

### CODE REVIEW

**Comment by AdrieanKhisbe** [comment]
(rebased)

File: packages/serverless-offline-kinesis/src/index.js [hunk 1]
```diff
@@ -12,22 +12,21 @@ const {
...
 
-const fromCallback = fun =>
-  new Promise((resolve, reject) => {
-    fun((err, data) => {
-      if (err) return reject(err);
-      resolve(data);
-    });
-  });
+const NO_KINESIS_FOUND = 'Could not find kinesis stream';
+const KINESIS_RETRY_DELAY = 2000;
```
Comment (godu) [inline packages/serverless-offline-kinesis/src/index.js]:
```suggestion
const KINESIS_RETRY_DELAY = 200;
```

File: packages/serverless-offline-kinesis/src/index.js [hunk 2]
```diff
@@ -128,46 +129,56 @@ class ServerlessOfflineKinesis {
...
+
+    if (kinesisStream instanceof Error) {
+      if (retry) {
+        this.serverless.cli.log(
+          `${streamName} - not found because of ${
+            kinesisStream.code
+          }, retrying in ${KINESIS_RETRY_DELAY}ms`
+        );
+        setTimeout(
+          this.createKinesisReadable.bind(this),
```
Comment (godu) [inline packages/serverless-offline-kinesis/src/index.js]:
```js
setTimeout(() => this.createKinesisReadable(....)
```

**Comment by AdrieanKhisbe** [comment]
(might need too rewrite it with just the retry)

**Comment by AdrieanKhisbe** [comment]
Superseeded by #140

**Pull request closed without merging**